sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -2085,4 +2087,28 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils with Te
       }
     }
   }
+
+  test("SPARK-28431: prevent CSV datasource throw TextParsingException with large size message") {
+    withTempDir { dir =>
+      val maxCharsPerCol = 10000
+      val str = "a" * (maxCharsPerCol + 1)
```

Comment:
Nit: seems we don't need to +1

Reply:
It do need "+1". Without "+1" won't throw exception. Because it do not exceed "max" value configured.

Reply:
Ah, yes. I thought the limit was not inclusive..